### PR TITLE
Incinerator multiblock

### PR DESCRIPTION
**Pull request by The-Minecraft-Scientist**
This provides a basic code implementation (and multiblock structure) for #285. 
Tentatively called the Incinerator but will likely be renamed to whatever name fits the final structure.
Currently the multi does not consume power, but could be modified to do so if needed.

Points to work out

- [ ] Decide on a final structure/design (current one is more to avoid bikeshedding than anything else)
- Balance
  - [x] Should it use power?
  - [ ] How expensive should it be? (related to structure/design)
  - [ ] How will the tiering (if any) work?
- [ ] Remove as many remaining automated item voiding methods as possible

(this PR also renames `VoidingMultiblockBase` to`FluidVoidingMultiblockBase` to fit in with `ItemVoidingMultiblockBase`).

FYI, I have done zero 1.12 modding, so please let me know if I footgun myself/otherwise write bad/wrong code

### CODE REVIEW

File: src/main/java/supersymmetry/common/metatileentities/multi/ItemVoidingMultiblockBase.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    private int actualVoidingRate() {
+        return baseVoidingRate() * this.bonusMultiplier;
+    }
+    protected void doVoiding() {
+        int toSubtract = actualVoidingRate();
+        for(IItemHandlerModifiable storage: getAbilities(MultiblockAbility.IMPORT_ITEMS)) {
+            for(int i = 0; i < storage.getSlots() && toSubtract > 0; i++) {
+                ItemStack stack = storage.getStackInSlot(i);
+                if(!stack.isEmpty()) {
+                    ItemStack replaceWith = stack.copy();
```
Comment (The-Minecraft-Scientist) [inline src/main/java/supersymmetry/common/metatileentities/multi/ItemVoidingMultiblockBase.java:31]: It feels like there should be a better way to do this, copying the ItemStack here seems excessive

Comment (bruberu) [reply src/main/java/supersymmetry/common/metatileentities/multi/ItemVoidingMultiblockBase.java:31]: You should be able to just extract the items from the inventory, I would imagine.

**Comment by The-Minecraft-Scientist** [comment]
Current state:
Multiblock does not consume power. gated behind aluminium (frame boxes).
It voids one item per tick for free, and up to an unlimited number of items per tick with boosting

You can boost the incinerator by giving it all of the inputs of a Large Gas Turbine recipe (just no lubricant). When boosted, the incinerator voids additional items every cycle (in addition to the free 1 item per tick) at a rate of one item per 10,000 EU generated by the fuel (note that oxidizers are still taken into account, it uses the SuSy LGT recipe map to compute this eu value). It will currently consume as much fuel as possible each cycle (essentially uncapping the boosted voiding rate), but that could easily be capped if there are balance concerns (in addition to changing the EU cost of voiding one item)

**Comment by The-Minecraft-Scientist** [comment]
Going to close this PR because the direction that was decided on for item voiding doesn't align with the code very well.

**Pull request closed without merging**